src/BinSkim.Driver/RoslynAnalyzer/ILOperations.cs

```diff
@@ -0,0 +1,485 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.CodeAnalysis.Semantics;
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.IL
+{
+    // Things that are needed from a Compilation to ensure various invariants.
+    internal interface ISymbolProvider
+    {
+        ITypeSymbol CreatePointerType(ITypeSymbol pointedAtType);
+
+        // FEEDBACK: Do we need to support custom bounds? (Not supported in VB/C#)
```

Comment:
Either we support them, in which case analyzer authors have to deal with things that are impossible in our source languages, or we refuse to import/analyze code that includes them. This will need discussion by a design team.